### PR TITLE
fix(#1406): allow nvim-tree.renderer.icons.show.folder_arrow

### LINKED ISSUE #1406: Disabling folder icons disables folder arrows
### Description

As discussed in #1405, setting,
```lua
renderer = {
  icons = {
    show = {
      file = true,
      folder = false,
      folder_arrow = true,
      git = true
    }
  }
}
```
Disables the arrows as well as the icons, whereas the expectation is that arrows would continue to be rendered.

### Neovim version

```text
NVIM v0.7.2
Build type: Release
LuaJIT 2.1.0-beta3
```

### Operating system and version

Linux 5.18.9-arch1-1

### nvim-tree version

4bd919a

### Minimal config

```lua
renderer = {
icons = {
  show = {
    file = true,
    folder = false,
    folder_arrow = true,
    git = true
  }
}
```

### Steps to reproduce

1. Modify config as above.
2. Observe lack of arrows rendered.

### Expected behavior

Folder arrows shown, no folder icons.

### Actual behavior

No folder arrows shown, no folder icons.

### PR DESCRIPTION
**Pull request by alex-courtis**
closes #1406

Always allows folder arrows. When using indent markers the arrows do not indent the surrounding nodes.

@milessabin please test the combinations of 
* renderer.icons.show.folder
* renderer.icons.show.folder_arrow
* renderer.indent_markers.enable

### CODE REVIEW

**Comment by kyazdani42** [comment]
The highlighting is not working as expected for the first iteration without the indent markers:

File: lua/nvim-tree/renderer/components/padding.lua [hunk 1]
```diff
@@ -1,21 +1,8 @@
...
-    if node.nodes then
-      local icon = M.config.icons.glyphs.folder[node.open and "arrow_open" or "arrow_closed"]
-      return string.rep(" ", depth - 2) .. icon .. " "
-    end
-    return string.rep(" ", depth)
-  end
-end
-
-local function get_padding_indent_markers(depth, idx, nodes_number, _, markers)
+local function get_padding_indent_markers(depth, idx, nodes_number, markers)
```
Comment (kyazdani42) [inline lua/nvim-tree/renderer/components/padding.lua:3]: signature of function should not be changed i believe.

**Comment by milessabin** [comment]
@alex-courtis LGTM! Many thanks for this!

**Review by alex-courtis — COMMENTED** [review]
Good catch for the highlighting.

**Pull request merged**